### PR TITLE
TUD API module initial commit

### PR DESCRIPTION
**Pull request by bramleyjl**
## Description of change
Initial PR for the test-user-dashboard API that will support the TUD frontend app. This change adds the `test-user-dashboard` engine, updates the vets-api schema to include the a new `test_user_dashboard_tud_accounts` table that will contain test user account information as well as the first draft of the master list of existing test accounts to be included, and contains routing for an tud_accounts index call

## Original issue(s)
https://app.zenhub.com/workspaces/vsp-identity-5f5bab705a94c9001ba33734/issues/department-of-veterans-affairs/va.gov-team/17208
https://app.zenhub.com/workspaces/vsp-identity-5f5bab705a94c9001ba33734/issues/department-of-veterans-affairs/va.gov-team/17214

## Things to know about this PR

The TUD backend is provided as a rails module and its routes will be accessible after running `bundle install`, the `tud_accounts` resource test route can be called at `localhost:3000/test_user_dashboard/tud_accounts`. TUD migrations are configured to be pulled into the main vets-api application's migration management and can be run from the application root.

After updating the vets-api DB schema the .csv of tud_accounts can be populated into it by running the new rake task `rake test_user_account:load`

Manual testing has been done to verify migrations, population script, and index route work correctly, unit tests have been added for the initial controller and service methods.

### CODE REVIEW

**Comment by LindseySaari** [comment]
Was this generated using the module generator? If so, we are asking that you keep the initial commit message, as we are using it for success metric and analytical purposes.

File: db/schema.rb
Comment (LindseySaari) [inline db/schema.rb:13]: It looks like dangerbot has flagged this PR for "Modified files in db/ and app/ inside the same PR!Modified files in db/ and app/ inside the same PR!"

Comment (bramleyjl) [reply db/schema.rb]: DB migrations have been removed and will be restored in a separate PR after this lands

**Comment by bramleyjl** [comment]
> Was this generated using the module generator? If so, we are asking that you keep the initial commit message, as we are using it for success metric and analytical purposes.

I updated the original commit message to `Initial commit of module structure *KEEP THIS COMMIT MESSAGE*`, is that the one you're referring to?

File: modules/test_user_dashboard/app/helpers/test_user_dashboard/tud_accounts_helper.rb
Comment (LindseySaari) [inline modules/test_user_dashboard/app/helpers/test_user_dashboard/tud_accounts_helper.rb:4]: Is this just a placeholder file for now?

Comment (bramleyjl) [reply modules/test_user_dashboard/app/helpers/test_user_dashboard/tud_accounts_helper.rb:4]: It was autogenerated when the `TudAccounts` resource was created, should it be removed since it's just a placeholder?

Comment (LindseySaari) [reply modules/test_user_dashboard/app/helpers/test_user_dashboard/tud_accounts_helper.rb:4]: I would say yes. I took a glance at existing modules and it looks like they don't have a helpers dir/placeholder files

**Review by LindseySaari — APPROVED** [review]
This looks good. I'll leave it up to you to remove the placeholder files or not

**Pull request merged**
